### DISCUSSION TITLE
Filtrage des données en utilisant une exclusion

### DISCUSSION THREAD

**Discussion by camilledesalabert** [Inist-CNRS/lodex#2046] [Ideas]
Actuellement, il est possible de filtrer un tableau de données via certains champs (ou colonnes) en recherchant une chaîne de caractères donnée (opérateur "contient"). Il serait utile de pouvoir également filtrer en excluant une chaîne de caractères (opérateur "ne contient pas"). Cela pourrait être intéressant lorsque l'on adapte un loader à des données et ainsi savoir par exemple quelles langues sont présentes dans un corpus avant d'avoir appliquer un modèle et publier les données.

**Comment by touv** [comment]
coté backoffice

**Comment by AnaelKremer** [comment]
on peut facilement inverser un includes :  

`value = get("value.authors").filter(author => author.rnsr.includes("198822446E"))`
ce code ne conserve que les auteurs ayant le rnsr "198822446E"

Pour inverser le filtre il suffit d'ajouter `!`
`value = get("value.authors").filter(author => !author.rnsr.includes("198822446E"))`
 
Là on ne conserve que les auteurs n'ayant **pas** ce rnsr.

Voir https://github.com/AnaelKremer/Atelier-Lodash-usage-Lodex/blob/f1aa3d1acf79e4d38aebae655ed1f76fff7f4e73/07-collections.md#filter
